napari/view_layers.py

```diff
@@ -1,703 +1,106 @@
-        Translation values for the layer.
-    opacity : float
-        Opacity of the layer visual, between 0.0 and 1.0.
-    blending : str
-        One of a list of preset blending modes that determines how RGB and
-        alpha values of the layer visual get mixed. Allowed values are
-        {'opaque', 'translucent', and 'additive'}.
-    visible : bool
-        Whether the layer visual is currently being displayed.
-    title : string
-        The title of the viewer window.
-    ndisplay : {2, 3}
-        Number of displayed dimensions.
-    order : tuple of int
-        Order in which dimensions are displayed where the last two or last
-        three dimensions correspond to row x column or plane x row x column if
-        ndisplay is 2 or 3.
-    axis_labels : list of str
-        Dimension names.
-    show : bool, optional
-        Whether to show the viewer after instantiation. by default True.
+    layer_string : str
+        The name of the layer type
 
     Returns
     -------
-    viewer : :class:`napari.Viewer`
-        The newly-created viewer.
-
-    Notes
-    -----
-    See vispy's marker visual docs for more details:
-    http://api.vispy.org/en/latest/visuals.html#vispy.visuals.MarkersVisual
+    view_func : Callable
+        The complete view_* function
     """
-    viewer = Viewer(
-        title=title,
-        ndisplay=ndisplay,
-        order=order,
-        axis_labels=axis_labels,
-        show=show,
+    add_string = f'add_{layer_string}'  # name of the corresponding add_* func
+    try:
+        add_method = getattr(AddLayersMixin, add_string)
+    except AttributeError:
+        raise AttributeError(f"No Viewer method named '{add_string}'")
+
+    # get signatures of the add_* method and Viewer.__init__
+    add_sig = inspect.signature(add_method)
+    viewer_sig = inspect.signature(Viewer)
+
+    # create a new combined signature
+    new_params = list(add_sig.parameters.values())[1:]  # [1:] to remove self
+    new_params += [
+        p.replace(kind=p.KEYWORD_ONLY) for p in viewer_sig.parameters.values()
+    ]
+    combo_sig = add_sig.replace(parameters=new_params)
+
+    # define the actual function that will create a new Viewer and add a layer
+    def real_func(*args, **kwargs):
+        view_kwargs = {}
+        for key in list(kwargs.keys()):
+            if key in viewer_sig.parameters:
+                view_kwargs[key] = kwargs.pop(key)
+        viewer = Viewer(**view_kwargs)
+        getattr(viewer, add_string)(*args, **kwargs)
+        return viewer
+
+    inner_sig = ", ".join([f"{p}={p}" for p in combo_sig.parameters])
+
+    # compile a new function with the combined signature that calls real_func
+    fakefunc = f"def func{combo_sig}:\n    return real_func({inner_sig})\n"
+    view_func_code = compile(fakefunc, "fakesource", "exec")
+
+    # evaluate the new function in a fake namespace and extract it
+    fakeglobals = {}
+    eval(view_func_code, {"real_func": real_func}, fakeglobals)
```

Comment:
This line is really the meat of it.  Here's how it works:

1. we define `real_func`, which is the (easier to understand) function that will do the work of creating a new viewer and adding a layer to it.
2. we create a **string** (`fakefunc`) that represents how we _would_ have typed out the original `view_*` method. 
    - `{combo_sig}` is an [inspect.Signature](https://docs.python.org/3/library/inspect.html#inspect.Signature) object (whose string representation is, conveniently, exactly how we would have typed the original function).
    - the inner `real_func({inner_sig})` part is basically how we were typing the body of the `view_*` functions before, e.g.: `(data=data, name=name, scale=scale ...)`
3. we compile that string into `view_func_code`
4. finally, we actually evaluate the compiled code in a (safe) empty namespace, and provide a `locals()` dict that tells python that the function name `real_func` in the `fakefunc` string actually corresponds to the `real_func` that we defined on line 66.   (Note: evaluation at this step is essentially exactly what was previously happening when python hit each `def view_*` declaration when importing `view_layers.py`)
5. we grab our newly evaluated function from the empty namespace that we supplied... and go on to add our documentation below